processing/src/main/java/org/apache/druid/query/groupby/GroupByQueryQueryToolChest.java

```diff
@@ -233,7 +235,7 @@ public GroupByQueryQueryToolChest(
         return groupByStrategy.processSubtotalsSpec(
             query,
             resource,
-            groupByStrategy.processSubqueryResult(subquery, query, resource, finalizingResults, false)
+            groupByStrategy.processSubqueryResult(subquery, withoutSubtotals(query), resource, finalizingResults, false)
```

Comment:
this needs more thinking. why weren't we passing `query.withSubtotalsSpec(null)` instead of `query`

Reply:
some context is in here but not helpful - https://github.com/apache/druid/pull/5280/files#r181885318
I am concerned that modifying the query cause some problem down the line.